demos/bert_question_answering_embedding_demo/python/bert_question_answering_embedding_demo.py

```diff
@@ -81,286 +87,243 @@ def build_argparser():
-            input_names_emb[2]: np.array([[0]     + [0] * num       + [0]     + tok_pad * pad_len], dtype=dtype),
-            input_names_emb[3]: np.arange(max_length, dtype=dtype)[None, :]
-        }
-
-        # calc embedding
-        ie_encoder_exec_emb = ie_encoder_exec_emb_dict[max_length]
-
-        t_start = time.perf_counter()
-        res = ie_encoder_exec_emb.infer(inputs=inputs)
-        t_end = time.perf_counter()
-        log.info("Embedding calculated for sequence of length {} with {:0.2f} requests/sec ({:0.2} sec per request)".format(
-            max_length,
-            1 / (t_end - t_start),
-            t_end - t_start
-        ))
-
-
-        res = res[output_names_model_emb[0]]
-        return res.squeeze(0)
-
-    #small class to store context as text and tokens and its embedding vector
-    class ContextData:
-        def __init__(self, context, c_tokens_id, c_tokens_se):
-            self.context = context
-            self.c_tokens_id = c_tokens_id
-            self.c_tokens_se = c_tokens_se
-            self.c_emb = calc_emb(self.c_tokens_id, max_length_c)
+    plugin_config = get_user_config(args.device, args.num_streams, args.num_threads)
 
-    paragraphs = get_paragraphs(args.input)
-    contexts_all = []
+    log.info('Reading Bert Embedding model {}'.format(args.model_emb))
+    model_emb = BertEmbedding(ie, args.model_emb, vocab, args.input_names_emb)
 
-    log.info("Indexing {} paragraphs...".format(len(paragraphs)))
-    for par in paragraphs:
-        c_tokens_id, c_tokens_se = text_to_tokens(par.lower(), vocab)
-        if not c_tokens_id:
-            continue
+    # reshape BertEmbedding model to infer short questions and long contexts
+    model_emb_exec_dict = {}
+    max_len_context = 384
+    max_len_question = 32
 
-        # get context as string and then encode it into token id list
-        # calculate number of tokens for context in each request.
-        # reserve 3 positions for special tokens
-        # [CLS] q_tokens [SEP] c_tokens [SEP]
-        if args.model_qa:
-            #to make context be able to pass model_qa together with question
-            c_wnd_len = max_length_qc - (max_length_q + 3)
+    for new_length in [max_len_question, max_len_context]:
+        model_emb.reshape(new_length)
+        if new_length == max_len_question:
+            model_emb_exec_dict[new_length] = ie.load_network(model_emb.net, args.device)
         else:
-            #to make context be able to pass model_emb without question
-            c_wnd_len = max_length_c - 2
+            model_emb_exec_dict[new_length] = AsyncPipeline(
+                ie, model_emb, plugin_config, device=args.device, max_num_requests=args.num_infer_requests
+            )
+    log.info('The Bert Embedding model {} is loaded to {}'.format(args.model_emb, args.device))
+    emb_pipeline = model_emb_exec_dict[max_len_context]
```

Comment:
Why do you define an alternative name for `model_emb_exec_dict[max_len_context]`? May be it is better to replace `for new_length in [max_len_question, max_len_context]` and create `emb_pipeline` and `emb_exec_net`.